common/lib/dependabot/errors.rb

```diff
@@ -4,7 +4,7 @@
 
 module Dependabot
   class DependabotError < StandardError
-    BASIC_AUTH_REGEX = %r{://(?<auth>[^:]*:[^@%\s]+(@|%40))}.freeze
+    BASIC_AUTH_REGEX = %r{://(?<auth>[^@%\s]+(@|%40))}.freeze
```

Comment:
See: https://github.com/dependabot/dependabot-core/pull/3327/files#diff-8310fd74ffea4abeb1ff784b465b75bc8e6b549319d4038b1516a0dc957e08b0R352

For Bundler 1, if we get either `Bundler::Fetcher::AuthenticationRequiredError` or `Bundler::Fetcher::BadAuthenticationError`, we re-wrap them as `Dependabot::PrivateSourceAuthenticationFailure` with the `source` extracted from the Bundler error.

In Bundler v2, in either scenario, we get a `Bundler::Fetcher::BadAuthenticationError` and can strip the `source` out using `BundlerErrorPatterns::BAD_AUTH_REGEX`.

The only caveat is that if the user set a username without a password, i.e. `https://my-malformed-creds@example.com/username` we would leave the `my-malformed-creds` in the error message since there was not a `:` preceding it.

I figured the least surprising behaviour is just to strip anything before @ in a URI, which results in the new test case above.

Reply:
> I figured the least surprising behaviour is just to strip anything before @ in a URI, which results in the new test case above.

This makes sense to me 👍

Reply:
Looks like this change broke a spec in common

Reply:
Pushed up a spec change

Reply:
I've walked back this change and just updated the test case to use a properly formed basic auth token.

This is a bit of a distraction to getting this merged as it seems this updated schema has the potential to mutate NPM dependencies which is why the npm tests start to fail.